kamelet-support/src/main/java/io/kaoto/backend/KamelPopulator.java

```diff
@@ -386,13 +387,15 @@ private FlowStep processStep(final Step step, final boolean to) {
     }
 
     public List<FlowStep> processSteps(final Branch b) {
-        var list = new LinkedList<FlowStep>();
+        final var list = new LinkedList<FlowStep>();
         if (b.getSteps() != null) {
-            for (Step step : b.getSteps()) {
-                if (step != null) {
-                    list.add(processStep(step, true));
-                }
-            }
+            b.getSteps().stream()
+                    //make sure we don't try to process null steps from the branch
+                    .filter(s -> !Objects.isNull(s))
+                    .map(s -> processStep(s, true))
+                    //make sure we don't add null steps, the unrecognized ones
+                    .filter(s -> !Objects.isNull(s))
```

Comment:
is this filter necessary ? is it possible that `processStep(s,true)` will be `null`  even if s isn't `null` ?